src/main/scala/mesosphere/util/state/MesosLeaderInfo.scala

```diff
@@ -0,0 +1,30 @@
+package mesosphere.util.state
+
+/**
+  * Utility class for keeping track of a Mesos Leader
+  */
+trait MesosLeaderInfo {
+
+  def currentLeaderUrl: Option[String]
+
+  def load(host: String, port: Int): Unit
```

Comment:
Why not directly set the full URL here? I think that future Mesos versions will support HTTPS (maybe it is in 0.23 already?).

I also do not like the method name "load" so much. Maybe `onAutoDetectedMasterUrl`? That tries to express that the implementation can choose to use this information or not.